### PR TITLE
[WEB-1933] refactor: link create/update for issues and modules

### PR DESCRIPTION
**Pull request by aaryan610**
#### Improvements:

1. Users can now enter links without the protocol. Just the domain and TLD will do.
2. Module links modal now doesn't close on error.

#### Refactor:

1. Moved link components from the `core` folder to the `modules` folder.
2. Updated the error messages for module link actions.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new URL validation method that enhances the handling of URLs across various components.
	- Implemented a new component for displaying a list of module links, improving the user interface.
	- Added a new modal for creating and updating module links, streamlining user interactions.
	- Introduced a new helper function for URL validation, improving the overall functionality of the application.

- **Bug Fixes**
	- Enhanced error handling for URL inputs, providing clearer feedback to users.

- **Documentation**
	- Improved documentation for new and modified functions, aiding developer understanding and usage. 

- **Chores**
	- Refactored several components for improved readability and maintainability.

### CODE REVIEW

**Review by pablohashescobar — APPROVED** [review]

File: web/core/components/modules/analytics-sidebar/root.tsx [hunk 1]
```diff
@@ -287,16 +268,18 @@ export const ModuleAnalyticsSidebar: React.FC<Props> = observer((props) => {
 
   return (
     <div className="relative">
-      <LinkModal
+      <CreateUpdateModuleLinkModal
         isOpen={moduleLinkModal}
         handleClose={() => {
           setModuleLinkModal(false);
-          setSelectedLinkToUpdate(null);
+          const timeoutId = setTimeout(() => {
```
Comment (rahulramesha) [inline web/core/components/modules/analytics-sidebar/root.tsx:275]: @aaryan610 Not so sure about this code, can you please explain why is it being cleared after a timer and why we are clearing this timeout after its use?

File: web/core/components/modules/links/create-update-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    handleSubmit,
+    control,
+    reset,
+  } = useForm<ModuleLink>({
+    defaultValues,
+  });
+
+  const onClose = () => {
+    handleClose();
+    const timeoutId = setTimeout(() => {
```
Comment (rahulramesha) [inline web/core/components/modules/links/create-update-modal.tsx:39]: here as well

File: web/core/components/modules/links/create-update-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      } else {
+        await updateLink(payload, data.id);
+        setToast({
+          type: TOAST_TYPE.SUCCESS,
+          title: "Success!",
+          message: "Module link updated successfully.",
+        });
+      }
+      onClose();
+      reset(defaultValues);
```
Comment (rahulramesha) [inline web/core/components/modules/links/create-update-modal.tsx:68]: Wont calling `onClose` reset to `defaultValues`?

File: web/core/components/modules/analytics-sidebar/root.tsx (hunk 1 above)
Comment (aaryan610) [reply web/core/components/modules/analytics-sidebar/root.tsx:275]: The input fields clear out even before the modal is closed, which gives a sense of flicker to the user. Hence, clearing them after the modal has closed.

File: web/core/components/modules/links/create-update-modal.tsx (hunk 3 above)
Comment (aaryan610) [reply web/core/components/modules/links/create-update-modal.tsx:68]: Yes, it should. I'll correct this.

File: web/core/components/modules/analytics-sidebar/root.tsx (hunk 1 above)
Comment (rahulramesha) [reply web/core/components/modules/analytics-sidebar/root.tsx:275]: Oh ok, but why do you need to clear the timeout?

**Review by rahulramesha — APPROVED** [review]

**Review by pushya22 — APPROVED** [review]

**Pull request merged**